### PR TITLE
Advanced horse/donkey/llama support

### PR DESCRIPTION
**Pull request by Moondarker**
1. Makes horse window name follow the same pattern in <1.14 and >=1.14 (window.type = EntityHorse and ChatMessage-like title)
2. Makes activateEntity follow the same pattern in <1.16 and >=1.16 (sneaking state sent in packet depends on bot's sneaking state)
3. Extends horse-like entities with all neccessary attributes (I'm not sure if you want it to be implemented this way, should I maybe move it into a separate object in entity or maybe don't modify entity at all? Will update the doc once this is clear)
4. Adds `bot.openHorse(entity)` allowing to reliably open horse-like entity inventories without mounting them

### CODE REVIEW

**Comment by Moondarker** [comment]
Also, it should make situation with #237 better, but if horses need any more work to be considered as "supported", please let me know :)

File: lib/plugins/horse.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const { callbackify } = require('../promise_utils')
+
+module.exports = inject
+
+function inject (bot) {
+  const metadataMap = {
+    base: {
```
Comment (rom1504) [inline lib/plugins/horse.js:7]: is this correct for all versions ?

Comment (Moondarker) [reply lib/plugins/horse.js:7]: Good call, looks like it is very likely not. Seems like another thing to be added in minecraft-data /._.

**Comment by rom1504** [comment]
reopen if you want to work on it again

**Pull request closed without merging**